diff --git a/arm/highfriction-primitiveall.world b/arm/highfriction-primitiveall.world
index 05c2f4f..f3d2401 100644
--- a/arm/highfriction-primitiveall.world
+++ b/arm/highfriction-primitiveall.world
@@ -41,8 +41,8 @@
         <solver>
           <type>quick</type>
           <min_step_size>0.0001</min_step_size>
-          <iters>50</iters>
-          <sor>1.3</sor>
+          <iters>500</iters>
+          <sor>1.0</sor>
         </solver>
         <constraints>
           <cfm>0.0</cfm>
diff --git a/arm/models/block-highfriction/model.sdf b/arm/models/block-highfriction/model.sdf
index 5d96bb1..fac8410 100644
--- a/arm/models/block-highfriction/model.sdf
+++ b/arm/models/block-highfriction/model.sdf
@@ -43,9 +43,9 @@
               <soft_cfm>0</soft_cfm>
               <soft_erp>0.2</soft_erp>
               <kp>1e12</kp>
-              <kd>1.0</kd>
+              <kd>1e8</kd>
               <max_vel>0.01</max_vel>
-              <min_depth>0.0</min_depth>
+              <min_depth>0.002</min_depth>
             </ode>
             <bullet>
               <soft_cfm>0</soft_cfm>
diff --git a/arm/models/ur10_schunk_hybrid-highfriction/model.sdf b/arm/models/ur10_schunk_hybrid-highfriction/model.sdf
index cfa7a87..c82d1a6 100644
--- a/arm/models/ur10_schunk_hybrid-highfriction/model.sdf
+++ b/arm/models/ur10_schunk_hybrid-highfriction/model.sdf
@@ -429,9 +429,9 @@
               <soft_cfm>0</soft_cfm>
               <soft_erp>0.2</soft_erp>
               <kp>1e12</kp>
-              <kd>1.0</kd>
+              <kd>1e8</kd>
               <max_vel>0.01</max_vel>
-              <min_depth>0.0</min_depth>
+              <min_depth>0.002</min_depth>
             </ode>
             <bullet>
               <soft_cfm>0</soft_cfm>
@@ -496,9 +496,9 @@
               <soft_cfm>0</soft_cfm>
               <soft_erp>0.2</soft_erp>
               <kp>1e12</kp>
-              <kd>1.0</kd>
+              <kd>1e8</kd>
               <max_vel>0.01</max_vel>
-              <min_depth>0.0</min_depth>
+              <min_depth>0.002</min_depth>
             </ode>
             <bullet>
               <soft_cfm>0</soft_cfm>
@@ -518,7 +518,7 @@
       <child>l_finger</child>
       <axis>
         <dynamics>
-          <damping>0.0</damping>
+          <damping>10.0</damping>
           <friction>0.0</friction>
         </dynamics>
         <xyz>1 0 0</xyz>
@@ -527,6 +527,11 @@
           <upper>0.014</upper>
         </limit>
       </axis>
+      <physics>
+        <ode>
+          <implicit_spring_damper>1</implicit_spring_damper>
+        </ode>
+      </physics>
     </joint>
 
     <joint name="r_finger_actuator" type="prismatic"> 
@@ -534,7 +539,7 @@
       <child>r_finger</child>
       <axis>
         <dynamics>
-          <damping>0.0</damping>
+          <damping>10.0</damping>
           <friction>0.0</friction>
         </dynamics>
         <xyz>1 0 0</xyz>
@@ -543,6 +548,11 @@
           <upper>0</upper>
         </limit>
       </axis>
+      <physics>
+        <ode>
+          <implicit_spring_damper>1</implicit_spring_damper>
+        </ode>
+      </physics>
     </joint>
     <!-- end: schunk mpg 80 hand -->
 
